### DISCUSSION TITLE
xroms: migrating metrics to derived variables

### DISCUSSION THREAD

**Discussion by tlsw231** [pydata/xarray#6603] [Q&A] [answered]
1. I have a dataset from an ocean model (ROMS).
2. I use `resample` to compute daily-averaged values of some variable.
3. I am trying to use `xroms.ddxi` to get the x-derivative of the daily-averaged variable with the code below (The variable `utemp_daily` is the daily-averaged variable and `utemp` is the original variable.):
```
xroms.ddxi(utemp_daily,utemp_daily.attrs['grid'])
```
5. I get the error message: `Unable to find any combinations of metrics for array dims {'xi_u', 'eta_rho', 's_w'} and axes ('Z',)"` 

I confirmed the daily-averaged variable does have a `grid` attribute.

Is there a way to seamless pass along the metrics to derived variables? 

Here is the output of `cf.describe()`:

![Screen Shot 2022-05-12 at 1 47 34 PM](https://user-images.githubusercontent.com/90293171/168147017-029b9a36-2814-4ac4-a7ce-2d750bcf6126.png)
.

**Comment by dcherian** [comment, marked as answer]
Hi @tlsw231 can you please move this question to either xgcm or xroms instead

**Reply by tlsw231** [reply]
Sure, thanks!
